### PR TITLE
UI changes according to support vertical flowchart and scrolling

### PR DESCRIPTION
**Pull request by utsavgarg**
This commit adds changes in accordance to #14 and issue #16. 
Changes as below:
- bigger size of boxes at the start
- flowchart extends vertically
- canvas scrolls using scroll wheel
- panning works as before by dragging canvas
- double click zooms and ctrl + double click shrinks
- '[' key zooms and ']' key shrinks
- tested for googleNet, resNet, alexNet, vgg and cascaded fcn
- made all buttons equal sized and added border radius
- also added tooltips on hover

### CODE REVIEW

File: ide/static/css/style.css [hunk 1]
```diff
@@ -73,6 +73,22 @@ html,body,#app,.app{
   padding-right: 40px;
 }
 
+.rhTooltip{
+  padding: 0px;
+}
+
```
Comment (virajprabhu) [inline ide/static/css/style.css]: Remove extra line

File: ide/static/js/layer.js [hunk 2]
```diff
@@ -25,6 +25,10 @@ class Layer extends React.Component {
         }}
         data-type={this.props.type}
         onClick={(event) => this.props.click(event, this.props.id)}
+        onMouseEnter={(event) => this.props.hover(event, this.props.id)}
+        data-tip
+        data-for='getContent'
+
```
Comment (virajprabhu) [inline ide/static/js/layer.js]: Remove newline

File: ide/static/js/netLayout_vertical.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
+// Assumption: test differs from train in the source of input and
+// some additions of accuracy layer
+
+// designed to work with  alexNet, vgg, googleNet and resNet.
+// will work with all prototxts which follows our assumption
+// but the UI may not be clean.
+
+export default function(net){
+// map[x] = [y1, y2, y3]
```
Comment (virajprabhu) [inline ide/static/js/netLayout_vertical.js]: Remove unused code

File: ide/static/js/netLayout_vertical.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+// allocatePosition finds the closest position available to preferred position
+function allocatePosition(layerId, preferredPosition){
+  if (!map.hasOwnProperty(preferredPosition[1])) {
+    map[preferredPosition[1]] = [];
+  }
+  let positionsX = map[preferredPosition[1]];
+  if (positionsX.indexOf(preferredPosition[0]) != -1) {
+    let temp = preferredPosition[0], i=2;
```
Comment (virajprabhu) [inline ide/static/js/netLayout_vertical.js]: Add space around assignment operator.

File: ide/static/js/panZoom.js [hunk 5]
```diff
@@ -105,7 +112,7 @@ export default function() {
 
   if ("onmousewheel" in document) { panZoom.onmousewheel = mousewheel; }
```
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: We should allow pinch to zoom out.

File: ide/static/js/tooltip.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  constructor(props) {
+    super(props);
+  }
+ 
+  render() {
+    if (this.props.hoveredLayer) {
+      const params = [];
+      const props = [];
+      const layer = this.props.net[this.props.hoveredLayer];
+      //console.log(layer);
```
Comment (virajprabhu) [inline ide/static/js/tooltip.js]: Remove logs

**Comment by utsavgarg** [comment]
@virajprabhu I have made the initial boxes slightly smaller, changed the positioning to adapt to the screen size and done code clean up. I also added functionality for pinch to zoom-in/out but do not have a way to check if it works.

File: ide/static/js/content.js [hunk 7]
```diff
@@ -242,8 +260,8 @@ class Content extends React.Component {
     Object.keys(positions).forEach(layerId => {
       const layer = net[layerId];
       layer.state = {
-        top: `${250 + 50 * positions[layerId][1]}px`,
-        left: `${20 + 180 * positions[layerId][0]}px`,
+        top: `${height + 65 * positions[layerId][1]}px`,
+        left: `${width + 80 * positions[layerId][0]}px`,
```
Comment (virajprabhu) [inline ide/static/js/content.js]: I'm not sure I still follow. Why is the offset hardcoded? Still seems a little off.
screen shot 2017-05-13 at 9 32 56 am 3

File: ide/static/js/panZoom.js [hunk 8]
```diff
@@ -50,10 +56,15 @@ export default function() {
   };
 
   window.onmouseup = function() {
-  //panZoom.onmouseup = function() {
```
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: Delete line.

File: ide/static/js/panZoom.js [hunk 9]
```diff
@@ -68,13 +79,14 @@ export default function() {
       canvas.y += e.pageY - previousMousePosition.y;
       canvas.updateContainerPosition();
       previousMousePosition = { x: e.pageX, y: e.pageY };
-      //instance.repaintEverything();
```
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: Delete.

File: ide/static/js/panZoom.js [hunk 10]
```diff
@@ -85,15 +97,14 @@ export default function() {
...
+    canvas.style.transitionDuration = "0.1s";
     canvas.updateContainerPosition();
     canvas.updateContainerScale();
     current.zoom = zoom;
     instance.setZoom(canvas.scale);
-    //instance.repaintEverything();
   }
 
-  var mousewheel, lastMouseWheelEventTime = Date.now();
+  /*var mousewheel, lastMouseWheelEventTime = Date.now();
```
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: Delete.

File: ide/static/js/panZoom.js (hunk 5 above)
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: This works fine now.

File: ide/static/js/content.js (hunk 7 above)
Comment (utsavgarg) [inline ide/static/js/content.js]: The height and width are now based on the browser innerHeight and width, 65 and 80 are for relative spacing between nodes. Not quite sure why its still not centring for other resolutions.

**Comment by virajprabhu** [comment]
Changes incorporated in #67.

**Pull request closed without merging**